specification/eventgrid/Azure.Messaging.EventGrid/main.tsp

```diff
@@ -131,106 +131,124 @@ namespace AzureMessagingEventGrid {
-  }, OkResponse>;
-
+  @doc("Event Subscription Resource.")
+  @resource("eventsubscriptions")
+  @parentResource(TopicResource)
+  model EventSubscription {
+    @key
+    @doc("Event subscription name.")
+    @visibility("read")
+    eventSubscriptionName: string;
+  }
 
-  @doc("Publish Batch of Cloud Events to namespace topic.")
-  @route("/topics/{topicName}:publish", {shared: true})
-  @post op PublishBatchOfCloudEvents is Azure.Core.RpcOperation<{
-    @doc("content type")
-    @header("content-type")
-    contentType: "application/cloudevents-batch+json; charset=utf-8";
+  @doc("Publish operation Response")
+  model PublishResponse {}
 
-    @doc("Topic Name.")
-    @path
-    topicName: string;
 
-    @doc("Array of Cloud Events being published.")
-    @body
-    events: CloudEventEvent[];
-  }, OkResponse>;
-   
-  @doc("Receive Batch of Cloud Events from the Event Subscription.")
-  @route("/topics/{topicName}/eventsubscriptions/{eventSubscriptionName}:receive")
-  @post op ReceiveBatchOfCloudEvents is Azure.Core.RpcOperation<{
-    @doc("Topic Name.")
-    @path
-    topicName: string;
+  // POST https://{namespaceName}.{region}.eventgrid.azure.net/topics/{topicName}:publish?api-version={apiVersion}}
 
-    @doc("Event Subscription Name.")
-    @path
-    eventSubscriptionName: string;
+  @doc("Publish Single Cloud Event to namespace topic.")
+  @route("/", {shared: true})
+  @action("publish")
+  @post op PublishCloudEvent is Azure.Core.ResourceAction<
+    TopicResource,
+    {
+      @doc("content type")
+      @header("content-type")
+      contentType: "application/cloudevents+json; charset=utf-8";
+
+      @doc("Single Cloud Event being published.")
+      @body 
+      event: CloudEventEvent;
+    },
+    PublishResponse
+    >;
 
-    @doc("Max Events count to be received.")
-    @query
-    maxEvents?: int32 = 1;
+  @doc("Publish Batch of Cloud Events to namespace topic.")
+  @route("/", {shared: true})
+  @action("publish")
+  @post op PublishBatchOfCloudEvents is Azure.Core.ResourceAction<
+    TopicResource,
+    {
+      @doc("content type")
+      @header("content-type")
+      contentType: "application/cloudevents-batch+json; charset=utf-8";
+
+      @doc("Array of Cloud Events being published.")
+      @body
+      events: CloudEventEvent[];
+    },
+    PublishResponse
+    >;
+
+  // POST https://{namespaceName}.{region}.eventgrid.azure.net/topics/{topicName}/eventsubscriptions/{eventSubscriptionName}:receive?api-Version={apiVersion}&timeout=60&maxEvents={maxEvents}
 
-    @doc("Timeout value for receive operation in Seconds. Default is 60 seconds.")
-    @query
-    timeout?: int32 =60;
-  }, ReceiveResponse & CreatedResponse >;
+   
+  @doc("Receive Batch of Cloud Events from the Event Subscription.")
+  @action("receive")
+  @post op ReceiveBatchOfCloudEvents is Azure.Core.ResourceAction<
+    EventSubscription,
+    {
+      @doc("Max Events count to be received.")
+      @query
+      maxEvents?: int32 = 1;
```

Comment:
Question: what does maxEvent? represent?

will maxEvent be included in the final REST call even if user did not specify that and its value will be default = 1?

same for timeout

Reply:
"?" means optional. Yes, this is defining the _server defaults_, the python SDK will not send anything by default in the request, and will rely on the service to default the values.